### ISSUE TITLE
Cargo build fails because of shaderc-sys and os error 

### ISSUE THREAD

**Issue by cauvmou** [vulkano-rs/vulkano#1715]
# Template

I am trying to build the triangle example and I am running into an error with `shaderc-sys`. I have updated my PATH variable to include ninja and cmake.

*   Version of vulkano: 0.20
*   OS: Windows 10
*   GPU (the selected PhysicalDevice): NVIDIA GeForce RTX 2080
*   GPU Driver: NVIDIA Game Ready Driver 30.0.14.7111
Cargo.toml:
```
[package]
name = "RustVulkan"
version = "0.1.0"
edition = "2018"

[dependencies]
vulkano="0.20"
vulkano-win="0.20"
winit="0.24"
vulkano-shaders = "0.20"
```

# Issue
I am getting the following error:
```
error: failed to run custom build command for shaderc-sys v0.6.3

Caused by:
  process didn't exit successfully: D:\VSC\RustVulkan\target\debug\build\shaderc-sys-641568c1b34185e6\build-script-build (exit code: 101)
  --- stdout
  cargo:warning=System installed library not found.  Falling back to build from source
  running: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\Common7\\IDE\\CommonExtensions\\Microsoft\\CMake\\CMake\\bin" "C:\\Users\\User\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\shaderc-sys-0.6.3\\build" "-G" "Ninja" "-DCMAKE_POSITION_INDEPENDENT_CODE=ON" "-DSPIRV_SKIP_EXECUTABLES=ON" "-DSPIRV_WERROR=OFF" "-DSHADERC_SKIP_TESTS=ON" "-DCMAKE_INSTALL_LIBDIR=lib" "-DCMAKE_C_FLAGS= /nologo /EHsc /MD" "-DCMAKE_CXX_FLAGS= /nologo /EHsc /MD" "-DCMAKE_C_FLAGS_RELEASE= /nologo /EHsc /MD" "-DCMAKE_CXX_FLAGS_RELEASE= /nologo /EHsc /MD" "-DSHADERC_ENABLE_SHARED_CRT=ON" "-DCMAKE_INSTALL_PREFIX=D:\\VSC\\RustVulkan\\target\\debug\\build\\shaderc-sys-5a0c647c08a093c2\\out" "-DCMAKE_C_COMPILER=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.26.28801/bin/HostX64/x64/cl.exe" "-DCMAKE_CXX_COMPILER=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.26.28801/bin/HostX64/x64/cl.exe" "-DCMAKE_ASM_FLAGS= -nologo -MD -Brepro" "-DCMAKE_ASM_COMPILER=C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.26.28801/bin/HostX64/x64/cl.exe" "-DCMAKE_BUILD_TYPE=Release"        

  --- stderr
  thread 'main' panicked at '
  failed to execute command: Access is denied. (os error 5)

  build script failed, must exit now', C:\Users\User\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.45\src\lib.rs:894:5
  note: run with RUST_BACKTRACE=1 environment variable to display a backtrace
```
I cannot find any resources on this particular error message.

**Comment by Rua** [comment]
Does the backtrace give any more information?

**Comment by cauvmou** [comment]
No, unfortunately the backtrace does not deliver any other useful information
This is the backtrace:
```
stack backtrace:
     0:     0x7ff755e28ebf - std::backtrace_rs::backtrace::dbghelp::trace
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\..\..\backtrace\src\backtrace\dbghelp.rs:98
     1:     0x7ff755e28ebf - std::backtrace_rs::backtrace::trace_unsynchronized
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\..\..\backtrace\src\backtrace\mod.rs:66
     2:     0x7ff755e28ebf - std::sys_common::backtrace::_print_fmt
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\sys_common\backtrace.rs:67
     3:     0x7ff755e28ebf - std::sys_common::backtrace::_print::{{impl}}::fmt
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\sys_common\backtrace.rs:46
     4:     0x7ff755e3f4fa - core::fmt::write
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\core\src\fmt\mod.rs:1110
     5:     0x7ff755e24948 - std::io::Write::write_fmt<std::sys::windows::stdio::Stderr>
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\io\mod.rs:1588
     6:     0x7ff755e2bfe6 - std::sys_common::backtrace::_print
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\sys_common\backtrace.rs:49
     7:     0x7ff755e2bfe6 - std::sys_common::backtrace::print
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\sys_common\backtrace.rs:36
     8:     0x7ff755e2bfe6 - std::panicking::default_hook::{{closure}}
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\panicking.rs:208
     9:     0x7ff755e2bac9 - std::panicking::default_hook
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\panicking.rs:225
    10:     0x7ff755e2c645 - std::panicking::rust_panic_with_hook
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\panicking.rs:622
    11:     0x7ff755e2c231 - std::panicking::begin_panic_handler::{{closure}}
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\panicking.rs:519
    12:     0x7ff755e2980f - std::sys_common::backtrace::__rust_end_short_backtrace<closure-0,!>
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\sys_common\backtrace.rs:141
    13:     0x7ff755e2c189 - std::panicking::begin_panic_handler
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\panicking.rs:515
    14:     0x7ff755e452cc - std::panicking::begin_panic_fmt
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\panicking.rs:457
    15:     0x7ff755d92bd1 - cmake::fail
                                 at C:\Users\User\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.45\src\lib.rs:894
    16:     0x7ff755d92464 - cmake::run
                                 at C:\Users\User\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.45\src\lib.rs:869
    17:     0x7ff755d8ec43 - cmake::Config::build
                                 at C:\Users\User\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.45\src\lib.rs:707
    18:     0x7ff755d83409 - build_script_build::build_shaderc_msvc
                                 at C:\Users\User\.cargo\registry\src\github.com-1ecc6299db9ec823\shaderc-sys-0.6.3\build\build.rs:111
    19:     0x7ff755d85ee6 - build_script_build::main
                                 at C:\Users\User\.cargo\registry\src\github.com-1ecc6299db9ec823\shaderc-sys-0.6.3\build\build.rs:311
    20:     0x7ff755d8927b - core::ops::function::FnOnce::call_once<fn(),tuple<>>
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\library\core\src\ops\function.rs:227
    21:     0x7ff755d8918b - std::sys_common::backtrace::__rust_begin_short_backtrace<fn(),tuple<>>
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\library\std\src\sys_common\backtrace.rs:125
    22:     0x7ff755d87ee1 - std::rt::lang_start::{{closure}}<tuple<>>
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\library\std\src\rt.rs:49
    23:     0x7ff755e2ca5a - core::ops::function::impls::{{impl}}::call_once
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\library\core\src\ops\function.rs:280
    24:     0x7ff755e2ca5a - std::panicking::try::do_call
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\panicking.rs:401
    25:     0x7ff755e2ca5a - std::panicking::try
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\panicking.rs:365
    26:     0x7ff755e2ca5a - std::panic::catch_unwind
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\panic.rs:434
    27:     0x7ff755e2ca5a - std::rt::lang_start_internal
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\/library\std\src\rt.rs:34
    28:     0x7ff755d87ebf - std::rt::lang_start<tuple<>>
                                 at /rustc/a178d0322ce20e33eac124758e837cbd80a6f633\library\std\src\rt.rs:48
    29:     0x7ff755d86996 - main
    30:     0x7ff755e44074 - invoke_main
                                 at d:\A01\_work\6\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
    31:     0x7ff755e44074 - __scrt_common_main_seh
                                 at d:\A01\_work\6\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
    32:     0x7ffd56d47034 - BaseThreadInitThunk
    33:     0x7ffd56e82651 - RtlUserThreadStart
```

**Comment by Rua** [comment]
Well, it does tell me that the problem is in the `shaderc-rs` library and not in Vulkano. You also seem to be using an old Vulkano version, see if the problem still occurs with 0.25?

**Comment by cauvmou** [comment]
I am using older versions because the example uses these old versions, updating the versions breaks the project. As some imports cannot be resolved anymore. In vulkano version 0.25.0 the `RenderPassAbstract` trait seems to be lost, same thing for `DeviceExtensions`

**Comment by Rua** [comment]
Ok, but how do you expect Vulkano to fix your issue then?

**Comment by Eliah-Lakhin** [comment]
@cauvmou There were a lot of changes since `0.20`. A number of things had been changed and removed. In general, API became simpler and easier to use(I hope). That's why, for example `RenderPassAbstract` is no longer needed. It's functionality is now covered implicitly and/or through the API static structs. So, if possible I would recommend to try to migrate to the latest version(ideally to the current `master`, which is going to be released to version `0.26` soon). Also, I hope that https://github.com/vulkano-rs/vulkano/issues/1527 migration notes will be handy too.

If migration is not feasible by some reasons, perhaps downgrading of Rust compiler could help in your situation. I remember there were some strange issues with Shaderc-rs at some point in the past and as far as I remember they were tied to something related to the Rust compiler. They didn't exist with the older version of Rust and they have gone with newer, but existed somewhere in between.

**Comment by cauvmou** [comment]
@Eliah-Lakhin Thank you, I think it is just weird that the example won't compile, I will look into newer versions.
